graylog2-server/src/main/java/org/graylog2/shared/journal/KafkaJournal.java

```diff
@@ -794,6 +803,31 @@ public Object apply(LogSegment segment) {
             return deletedSegments;
         }
 
+        /** Change the load balancer status from ALIVE to THROTTLE, or vice versa depending on the
+         * journal utilization percentage. As the utilization ratio is reliable only after cleanup,
+         * that's where this is called from.
+         */
+        private void updateLoadBalancerStatus(double utilizationPercentage) {
```

Comment:
I'm a little bit uncomfortable with changing the server/load balancer status directly from this class.
An alternative could be emitting an event on the `EventBus` and a subscriber to that event which will change the `serverStatus` accordingly.